### DISCUSSION TITLE
Build Failed

### DISCUSSION THREAD

**Discussion by andybantly** [obsproject/obs-studio#4486] [Development Questions] [answered]
Source downloaded on 4/8/2021
I'm getting these build errors
1>winrt-capture.cpp
1>C:\Users\andyb\Documents\Visual Studio 2019\Projects\obs-studio-master\libobs-winrt\winrt-capture.cpp(377,11): error C2039: 'IsCursorCaptureEnabled': is not a member of 'winrt::Windows::Graphics::Capture::GraphicsCaptureSession'
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt/impl/Windows.Graphics.Capture.2.h(55): message : see declaration of 'winrt::Windows::Graphics::Capture::GraphicsCaptureSession'
1>C:\Users\andyb\Documents\Visual Studio 2019\Projects\obs-studio-master\libobs-winrt\winrt-capture.cpp(461,11): error C2039: 'IsCursorCaptureEnabled': is not a member of 'winrt::Windows::Graphics::Capture::GraphicsCaptureSession'
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt/impl/Windows.Graphics.Capture.2.h(55): message : see declaration of 'winrt::Windows::Graphics::Capture::GraphicsCaptureSession'
1>C:\Users\andyb\Documents\Visual Studio 2019\Projects\obs-studio-master\libobs-winrt\winrt-capture.cpp(607,21): error C2039: 'IsCursorCaptureEnabled': is not a member of 'winrt::Windows::Graphics::Capture::GraphicsCaptureSession'
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.18362.0\cppwinrt\winrt/impl/Windows.Graphics.Capture.2.h(55): message : see declaration of 'winrt::Windows::Graphics::Capture::GraphicsCaptureSession'

I commented them out and the obs-studio solution built.

When I run the program I have no Video Capture Devices and my webcam works with other software.

I followed the recipe for configuring my machine with Qt, MSys2, CMake, and building of FFMpeg

What could be the reason for not having a Video Capture Device and is it related to the build errors?

![image](https://user-images.githubusercontent.com/24298392/114197002-8899a400-9917-11eb-95ed-e40a3d9c8839.png)

**Comment by notr1ch** [comment]
You need to update your Windows SDK and check win-dshow is building successfully.

**Reply by andybantly** [reply]
I have updated to the latest Windows SDK (10.0.19041.0)
I do not see win-dshow in the plugins folder for the CMake generated solution.

How can I get win-dshow to be part of the generated solution?  I did try adding a virtual cam guid in win-dshow's CMakeLists.txt and then generating the solution again.  That didn't have any effect.

**Reply by andybantly** [reply]
I found this in the CMake output
"libdshowcapture submodule not found!  Please fetch submodules.  win-dshow plugin disabled."

Where can I find libdshowcapture submodule?

**Comment by andybantly** [comment, marked as answer]
I solved the problem.  The issue is that there is no source code in 'libdshowcapture' when you download the project as a ZIP.  When you clone the project that source then exists. 'Video Capture Device' now shows up.  I believe there is an issue with incomplete source code when it is downloaded as a ZIP file.

**Reply by Fenrirthviti** [reply]
OBS uses several submodules, which is explained in our [build guide](https://github.com/obsproject/obs-studio/wiki/install-instructions#windows-build-directions) that you would be doing a recursive clone, not downloading the source as a zip from github. That is not a supported method.

Make sure you always check the project documentation.
